developer/services/qrexec3.md

```diff
@@ -180,173 +150,136 @@ By contrast, consider this command:
-device...
+Sometimes just service name isn't enough to make reasonable qrexec policy.
+One example of such a situation is [qrexec-based USB passthrough](https://github.com/qubesos/qubes-issues/issues/531) - using just service name isn't possible to express the policy "allow access to device X and deny to others".
+It also isn't feasible to create a separate service for every device...
 
-For this reason, starting with Qubes 3.2, it is possible to specify a service
-argument, which will be subject to policy. Besides the above example of USB
-passthrough, a service argument can make many service policies more fine-grained
-and easier to write precise policy with "allow" and "deny" actions, instead of
-"ask" (offloading additional decisions to the user). And generally the less
-choices the user must make, the lower the chance to make a mistake.
+For this reason, starting with Qubes 3.2, it is possible to specify a service argument, which will be subject to policy.
+Besides the above example of USB passthrough, a service argument can make many service policies more fine-grained and easier to write precise policy with "allow" and "deny" actions, instead of "ask" (offloading additional decisions to the user).
+And generally the less choices the user must make, the lower the chance to make a mistake.
 
-The syntax is simple: when calling a service, add an argument to the service name
-separated with `+` sign, for example:
+The syntax is simple: when calling a service, add an argument to the service name separated with `+` sign, for example:
 
     /usr/lib/qubes/qrexec-client-vm target_vm_name RPC_ACTION_NAME+ARGUMENT
 
-Then create a policy as usual, including the argument
-(`/etc/qubes-rpc/policy/RPC_ACTION_NAME+ARGUMENT`). If the policy for the specific
-argument is not set (file does not exist), then the default policy for this service
-is loaded (`/etc/qubes-rpc/policy/RPC_ACTION_NAME`).
+Then create a policy as usual, including the argument (`/etc/qubes-rpc/policy/RPC_ACTION_NAME+ARGUMENT`).
+If the policy for the specific argument is not set (file does not exist), then the default policy for this service is loaded (`/etc/qubes-rpc/policy/RPC_ACTION_NAME`).
 
 In target VM (when the call is allowed) the service file will searched as:
 
  - `/etc/qubes-rpc/RPC_ACTION_NAME+ARGUMENT`
  - `/etc/qubes-rpc/RPC_ACTION_NAME`
 
-In any case, the script will receive `ARGUMENT` as its argument and additionally as
-`QREXEC_SERVICE_ARGUMENT` environment variable. This means it is also possible
-to install a different script for a particular service argument.
+In any case, the script will receive `ARGUMENT` as its argument and additionally as `QREXEC_SERVICE_ARGUMENT` environment variable.
+This means it is also possible to install a different script for a particular service argument.
 
 See below for an example service using an argument.
 
-### Revoking "Yes to All" authorization ###
+### Revoking "Yes to All" authorization
```

Comment:
Technically it doesn't apply to R4.0 at all, because the current confirmation dialog doesn't have "yes to all" action.
There is a plan to restore this or similar functionality in R4.1, so maybe just add a comment?